src/main/java/ChatServer.java

```diff
@@ -0,0 +1,71 @@
+import java.io.*;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class ChatServer {
+    private ServerSocket serverSocket;
+    private UserIO io;
+
+    public ChatServer(ServerSocket serverSocket, UserIO io) {
+        this.serverSocket = serverSocket;
+        this.io = io;
+    }
+
+    public void readInFromClient() {
+        Socket server = makeConnection(serverSocket);
+        BufferedReader reader = createBufferedReader(server);
+        readInputTillOver(reader, server);
+        io.showExitMessage();
+        closeSocket();
+    }
+
+    private Socket makeConnection(ServerSocket serverSocket) {
+        try {
```

Comment:
All these try-catch blocks make my eyes bleed.

Although you have the right idea, you can actually get away (I think) with having one try-catch on the `public` method. `private` methods are only ever called within the class so it's not as if they are leaking out to clients of this class. It's only the `public` methods that need a try-catch. Always try to put try-catch's at the highest level possible; when it comes to de-checking exceptions you can always put them at the public method.